src/features/singleRight/delegate/ChooseRightsPage/ChooseRightsPage.tsx

```diff
@@ -57,6 +81,16 @@ export const ChooseRightsPage = () => {
     setSelectedRights(initialCheckedRightsList);
   }, []);
 
+  useMemo(() => {
```

Comment:
The useMemo listens to changes in processedDelegations and when there's a change it checks the processedDelegationsRatio

Reply:
Good point.
But then, why not use a useEffect then?
I don't see the need for memoization specifically if what is intended here is just something that triggers based on a change in a specific parameter.

Reply:
useMemo is faster than useEffect and it doesn't cause side effects, so I thought it was safer. However when I dig into it, it seems like useMemo should primarily be used for recomputing values. Also with useEffect I see that I don't need to delay the navigation to the receipt page to avoid state setting bugs and we can console.log stuff inside it if there's any problems with the functionality in the future.

So then I'll change to useEffect.